### DISCUSSION TITLE
higress获取不到本地k8s集群的nacos服务

### DISCUSSION THREAD

**Discussion by Harrybooo** [alibaba/higress#585] [Q&A]
我在本地创建了一个k8s的集群，使用helm安装了higress成功后，又在其中运行一个nacos的集群，在运行了一个java程序之后，它注册上了nacos并成功运行，但是在higress上设置路由的时候却找不到这个java服务，只有k8s的服务可以找到，服务来源也配置了，来源ip试过了集群ip和外部ip都还是获取不到，请问这是什么问题

**Comment by CH3CHO** [comment]
Nacos是什么版本，需要认证吗？服务来源和服务列表页面方便截图看一下吗？

**Reply by Harrybooo** [reply]
版本是2.1.2
![image](https://github.com/alibaba/higress/assets/134490336/1a8f6d38-a511-46d8-84f5-495e067529bd)

**Reply by Harrybooo** [reply]
我没有开启nacos的鉴权，这个认证是输入nacos网页端的账号密码吗？我认证打开与关闭都试了，还是发现不了服务

**Reply by CH3CHO** [reply]
检查一下controller pod的日志，另外这个IP的9848端口是通的吗？

**Reply by Harrybooo** [reply]
好的感谢，我检查一下

**Comment by Harrybooo** [comment]
![image](https://github.com/alibaba/higress/assets/134490336/8b5145a2-963d-4d0f-b36a-1e69da64a3a1)
nacos的9848端口是放开的openstack给集群的安全组也是放开的

**Comment by Harrybooo** [comment]
日志没有找到任何关于nacos的服务信息
![image](https://github.com/alibaba/higress/assets/134490336/1323f9aa-8e66-4e9d-96e6-93ed8fa5eabc)

**Reply by CH3CHO** [reply]
controller的话加上 -c higress-core 参数看一下日志

**Comment by Harrybooo** [comment]
![image](https://github.com/alibaba/higress/assets/134490336/b788385b-930a-43df-ab34-38942316588c)
一直在报这个错

**Reply by CH3CHO** [reply]
那就是pod里面连不上nacos服务。exec进去测一下吧。

**Comment by Harrybooo** [comment]
嘶，好的谢谢大佬

**Comment by Harrybooo** [comment]
![image](https://github.com/alibaba/higress/assets/134490336/cb726405-c547-482f-8cbb-6a9ad2e26220)
这能说明是通的吗（狗头）pod里面工具太少了

**Reply by CH3CHO** [reply]
再试一下9848看看。

**Reply by Harrybooo** [reply]
![image](https://github.com/alibaba/higress/assets/134490336/d7cc7b3a-f75d-4458-9efc-d47759c3991e)
curl: (1) Received HTTP/0.9 when not allowed

**Comment by Harrybooo** [comment]
![image](https://github.com/alibaba/higress/assets/134490336/00553eba-31c2-4bf4-87ed-8a99d71c3a1f)
请问这个是为什呢

**Reply by CH3CHO** [reply]
看不出什么，加上-svk参数再试试。另外也看看nacos那边有没有什么有用的日志。

**Comment by Harrybooo** [comment]
好的

**Comment by Harrybooo** [comment]
![image](https://github.com/alibaba/higress/assets/134490336/061afe4a-874b-4089-b742-afc8412ea541)
nacos的日志

**Comment by Harrybooo** [comment]
![image](https://github.com/alibaba/higress/assets/134490336/b9ad0b85-473f-4678-b0d9-6bd9684ec6a5)

**Reply by CH3CHO** [reply]
这怎么感觉nacos不太正常呢？要不弄个Java的客户端做个服务发现看看？

**Comment by Harrybooo** [comment]
nacos的日志稍等，我在仔细查一下

**Comment by Harrybooo** [comment]
谢谢大佬，我查看的是nacos集群确实有问题，重新安装排除问题之后，还是连接不上，之后重新安装了higress就可以了

**Reply by CH3CHO** [reply]
好的~
